src/Sylius/Component/Core/spec/Promotion/Checker/ItemsFromTaxonTotalRuleCheckerSpec.php

```diff
@@ -0,0 +1,161 @@
+    {
+        $this->beConstructedWith($taxonRepository);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Component\Core\Promotion\Checker\ItemsFromTaxonTotalRuleChecker');
+    }
+
+    function it_implements_rule_checker_interface()
+    {
+        $this->shouldImplement(RuleCheckerInterface::class);
+    }
+
+    function it_recognizes_subject_as_eligible_if_it_has_items_from_configured_taxon_which_has_required_total(
+        OrderInterface $order,
+        OrderItemInterface $item1,
+        OrderItemInterface $item2,
+        OrderItemInterface $item3,
+        ProductInterface $compositeBow,
+        ProductInterface $longsword,
+        ProductInterface $reflexBow,
+        TaxonInterface $bows,
+        TaxonRepositoryInterface $taxonRepository
+    ) {
+        $order
+            ->getItems()
+            ->willReturn(
+                new \ArrayIterator([$item1->getWrappedObject(), $item2->getWrappedObject(), $item3->getWrappedObject()])
+            )
+        ;
+
+        $taxonRepository->findOneBy(['code' => 'bows'])->willReturn($bows);
+
+        $item1->getProduct()->willReturn($compositeBow);
+        $compositeBow->hasTaxon($bows)->willReturn(true);
+        $item1->getTotal()->willReturn(5000);
+
+        $item2->getProduct()->willReturn($longsword);
+        $longsword->hasTaxon($bows)->willReturn(false);
+        $item2->getTotal()->willReturn(4000);
+
+        $item3->getProduct()->willReturn($reflexBow);
+        $reflexBow->hasTaxon($bows)->willReturn(true);
+        $item3->getTotal()->willReturn(9000);
+
+        $this->isEligible($order, ['taxon' => 'bows', 'amount' => 10000])->shouldReturn(true);
+    }
+
+    function it_recognizes_subject_as_eligible_if_it_has_items_from_configured_taxon_which_has_total_equal_with_required(
+        OrderInterface $order,
+        OrderItemInterface $item1,
+        OrderItemInterface $item2,
+        ProductInterface $compositeBow,
+        ProductInterface $reflexBow,
+        TaxonInterface $bows,
+        TaxonRepositoryInterface $taxonRepository
+    ) {
+        $order->getItems()->willReturn(new \ArrayIterator([$item1->getWrappedObject(), $item2->getWrappedObject()]));
+
+        $taxonRepository->findOneBy(['code' => 'bows'])->willReturn($bows);
+
+        $item1->getProduct()->willReturn($compositeBow);
+        $compositeBow->hasTaxon($bows)->willReturn(true);
+        $item1->getTotal()->willReturn(5000);
+
+        $item2->getProduct()->willReturn($reflexBow);
+        $reflexBow->hasTaxon($bows)->willReturn(true);
+        $item2->getTotal()->willReturn(5000);
+
+        $this->isEligible($order, ['taxon' => 'bows', 'amount' => 10000])->shouldReturn(true);
+    }
+
+    function it_does_not_recognize_subject_as_eligible_if_items_from_required_taxon_has_too_low_total(
+        OrderInterface $order,
+        OrderItemInterface $item1,
+        OrderItemInterface $item2,
```

Comment:
But since there is already `ProductInterface $compositeBow` it would be more readable to name associated item `OrderItemInterface $compositeBowItem` :)